lib/msal-core/src/AuthError.ts

```diff
@@ -0,0 +1,240 @@
+/**
+  * Copyright (c) Microsoft Corporation
+  *  All Rights Reserved
+  *  MIT License
+  *
+  * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  * software and associated documentation files (the "Software"), to deal in the Software
+  * without restriction, including without limitation the rights to use, copy, modify,
+  * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  * permit persons to whom the Software is furnished to do so, subject to the following
+  * conditions:
+  *
+  * The above copyright notice and this permission notice shall be
+  * included in all copies or substantial portions of the Software.
+  *
+  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+  * OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
+  * OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+  */
+
+import { Constants } from "./Constants";
+import { ErrorMessage } from "./ErrorMessage";
+import { Utils } from "./Utils";
+/**
+* @hidden
+*
+* General error class thrown by the MSAL.js library.
+*/
+export class AuthError extends Error {
+    error: string;
+    errorDesc: string;
+
+    constructor(error: string, detailMessage: string) {
+        super(detailMessage);
+        this.error = error;
+        this.errorDesc = detailMessage;
+        this.name = "AuthError";
+        this.stack = new Error().stack;
+
+        Object.setPrototypeOf(this, AuthError.prototype);
+    }
+    static createUnexpectedError(errDesc: string) {
+        return new AuthError(ErrorMessage.unexpectedError.code, errDesc);
+    }
+}
+
+/**
+ * @hidden
+ *
+ * Error thrown when there is an error in the client code running on the browser.
+ */
+export class ClientAuthError extends AuthError {
+    constructor(error: string, errorDesc: string) {
+        super(error, errorDesc);
+        this.name = "ClientAuthError";
+
+        Object.setPrototypeOf(this, ClientAuthError.prototype);
+    }
+
+    static createEndpointResolutionError(errDesc: string): ClientAuthError {
+        var errorMessage = "Error: could not resolve endpoints. Please check network and try again.";
+        if (!Utils.isEmpty(errDesc)) {
+            errorMessage += " Details: " + errDesc;
+        }
+        return new ClientAuthError(ErrorMessage.endpointResolutionError.code, errorMessage);
+    }
+
+    static createMultipleMatchingTokensInCacheError(scope: string): ClientAuthError {
+        return new ClientAuthError(ErrorMessage.multipleMatchingTokens.code,
+            "Cache error for scope " + scope + ": " + ErrorMessage.multipleMatchingTokens.desc);
```

Comment:
in all of this file, instead of using + signs to concat use string interpolation

`Cache error for scope ${scope}: ${ErrorMessage.multiplematchingTokens.desc}`;

Reply:
+1, string interpolation for the rest of the code should be coming in a separate PR.

Reply:
while you have error here why not just update this file use interpolation?

Reply:
I think @sameerag has these updates in a separate PR.